packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -0,0 +1,1279 @@
+				op.clientTag,
+				op.url,
+			);
+			state.clients.push(newClient);
+			return state;
+		}
+		if (isOperationType<RemoveClient>("removeClient", op)) {
+			const removed = state.clients.splice(
+				state.clients.findIndex((c) => c.tag === op.clientTag),
+				1,
+			);
+			removed[0].container.dispose();
+			return state;
+		}
+		return model.reducer(state, op);
+	};
+
+	return {
+		...model,
+		minimizationTransforms,
+		generatorFactory,
+		reducer,
+	};
+}
+
+/**
+ * Mixes in functionality to generate an 'attach' op, which
+ * @privateRemarks This is currently file-exported for testing purposes, but it could be reasonable to
+ * expose at the package level if we want to expose some of the harness's building blocks.
+ */
+function mixinAttach<TOperation extends BaseOperation, TState extends LocalServerStressState>(
+	model: LocalServerStressModel<TOperation, TState>,
+	options: LocalServerStressOptions,
+): LocalServerStressModel<TOperation | Attach, TState> {
+	const { numOpsBeforeAttach } = options.detachedStartOptions;
+	if (numOpsBeforeAttach === 0) {
+		// not wrapping the reducer/generator in this case makes stepping through the harness slightly less painful.
+		return model as LocalServerStressModel<TOperation | Attach, TState>;
+	}
+	const attachOp = async (): Promise<TOperation | Attach> => {
+		return { type: "attach" };
+	};
+
+	const generatorFactory: () => AsyncGenerator<TOperation | Attach, TState> = () => {
+		const baseGenerator = model.generatorFactory();
+		return chainAsync(
+			takeAsync(numOpsBeforeAttach, baseGenerator),
+			takeAsync(1, attachOp),
+			baseGenerator,
+		);
+	};
+
+	const minimizationTransforms = model.minimizationTransforms as
+		| MinimizationTransform<TOperation | Attach>[]
+		| undefined;
+
+	const reducer: AsyncReducer<TOperation | Attach, TState> = async (state, operation) => {
+		if (isOperationType<Attach>("attach", operation)) {
+			state.isDetached = false;
+			assert.equal(state.clients.length, 1);
+			const clientA: Client = state.clients[0];
+
+			await clientA.container.attach(createLocalResolverCreateNewRequest("stress test"));
+			const url = await clientA.container.getAbsoluteUrl("");
+			assert(url !== undefined, "container must have a url");
+			const clients: Client[] = await Promise.all(
+				Array.from({ length: options.numberOfClients }, async (_, index) =>
+					loadClient(
+						state.localDeltaConnectionServer,
+						state.codeLoader,
+						state.tag("client"),
+						url,
+					),
+				),
+			);
+
+			// While detached, the initial state was set up so that the 'summarizer client' was the same as the detached client.
+			// This is actually a pretty reasonable representation of what really happens.
+			// However, now that we're transitioning to an attached state, the summarizer client should never have any edits.
+			// Thus we use one of the clients we just loaded as the summarizer client, and keep the client around that we generated the
+			// attach summary from.
```

Comment:
```suggestion
			// After attaching, we use the initial client as a read-only client for consistency comparison.
			// This makes debugging easier as the state of a client is easier to interpret if it has no local changes.
```

since we don't actually use this for summarization explicitly anymore as that is managed by authentic summarization flow. Would be reasonable to audit other docs, though up to you if you want to change the property name.